classes/ImageType.php

```diff
@@ -112,6 +112,51 @@ public static function getImagesTypes($type = null, $orderBySize = false)
         return self::$images_types_cache[$type];
     }
 
+    /**
+     * Returns image type definitions by name.
+     *
+     * @param string $name name
+     * @param string|null $type Image type
+     * @param bool $orderBySize
+     *
+     * @return array Image type definitions
+     *
+     * @throws PrestaShopDatabaseException
+     */
+    public static function getImagesTypesByName($name, $type = null, $orderBySize = false)
+    {
+        if (!isset(self::$images_types_cache[$type])) {
```

Comment:
I don't understand this method, it seems like a duplicate of `ImageType::getImagesTypes` except you find the image type based on its name and not its type Why not I understand the need and the query is adapted, however you cache the image type the same way it's done in the other method based on its type For the cache to be effective shouldn't you store it based on its name? Or at least loop through the existing cache to find an image type matching the requested name?

Reply:
You're right, I can loop on the existing cache. I thought it was better to have a method to find image types by name, but I can loop, it will cost less.